src/components/AddEvent.tsx

```diff
@@ -117,31 +110,50 @@ const AddEvent = () => {
       const dateArray = [];
       let currentDate = new Date(startDate);
       while (currentDate <= new Date(stopDate)) {
-        dateArray.push(fromDateToString(currentDate));
-        currentDate = currentDate.addDays(1);
+        switch (whenToRecur) {
+          case "day":
+            dateArray.push(fromDateToString(currentDate));
+            currentDate = currentDate.addDays(1);
+            break;
+          case "workday":
+            if (currentDate.getDay() > 0 && currentDate.getDay() < 6) {
+              dateArray.push(fromDateToString(currentDate));
+            }
+            currentDate = currentDate.addDays(1);
+            break;
+          case "weekend":
+            if (currentDate.getDay() === 0 || currentDate.getDay() === 6) {
+              dateArray.push(fromDateToString(currentDate));
+            }
+            currentDate = currentDate.addDays(1);
+            break;
+        }
       }
       return dateArray;
     };
 
-    if (recurrence) {
+    if (recurrence !== "no") {
       const lastDate = new Date(date).addDays(parseInt(recurrence));
       const dates = getDates(date, fromDateToString(lastDate));
-      const newData = dates.map((elem, i) => {
-        if (!!data[i]?.event?.length) {
-          const eventArrayWithAlreadyExistingEvent = [...data[i].event];
-          eventArrayWithAlreadyExistingEvent.push(newEvent);
-          return {
-            date: elem,
-            event: eventArrayWithAlreadyExistingEvent,
-          };
-        } else {
-          return {
-            date: elem,
-            event: [newEvent],
-          };
-        }
+      const newData = dates.map((date) => {
```

Comment:
Try to refactor from line 138 to 155. Think of a shorter way. Is everything necessary?